library/std/src/sys/unix/thread.rs

```diff
@@ -131,6 +148,19 @@ impl Thread {
         }
     }
 
+    #[cfg(target_os = "fuchsia")]
+    pub fn set_name(name: &CStr) {
+        use self::zircon::*;
+        unsafe {
+            zx_object_set_property(
+                zx_thread_self(),
+                ZX_PROP_NAME,
+                name.as_ptr() as *const libc::c_void,
+                libc::strlen(name.as_ptr()),
```

Comment:
CStr's know their own length:
```suggestion
                name.to_bytes().len(),
```